ql/math/randomnumbers/fractionalnoisersg.hpp

```diff
@@ -0,0 +1,149 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2022 Chester Wong
+ Copyright (C) 2022 Wojciech Czernous
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file fractionalnoisersg.hpp
+    \brief Fractional Gaussian noise (sequence) generator
+*/
+
+#ifndef quantlib_fractional_gaussian_noise_rsg_hpp
+#define quantlib_fractional_gaussian_noise_rsg_hpp
+
+#include <ql/math/fastfouriertransform.hpp>
+#include <ql/math/matrix.hpp>
+#include <ql/methods/montecarlo/sample.hpp>
+#include <complex>
+
+using std::pow;
+
+namespace QuantLib {
+
+    //! Fractional Gaussian noise (sequence) generator
+    /*! Using the method by Davies & Harte, improved by Chan & Wood, of finding
+        the square root of the autocovariance matrix by means of embedding 
+        it in a circulant matrix.
+        -# Andrew T. A. Wood, & Chan, G. (1994). Simulation of Stationary 
+        Gaussian Processes in [0,1]^d. Journal of Computational and Graphical
+        Statistics, 3(4), 409–432. https://doi.org/10.2307/1390903
+
+        \test the correctness of the returned values is tested by 
+              a likelihood-ratio test (LRT) of covariance matrix 
+    */
+    template <class GSG>
+    class FractionalGaussianNoiseRsg {
+      public:
+        typedef Sample<std::vector<Real> > sample_type;
+        FractionalGaussianNoiseRsg(Real HurstParameter,
```

Comment:
Not a big deal, but the style in QuantLib is to use camel case for function arguments. See https://www.quantlib.org/style.shtml for reference.